pkg/controller/cache/cluster/controller.go

```diff
@@ -0,0 +1,201 @@
+	if err := c.client.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+	awsClient, err := c.newClientFn(ctx, s.Data[p.Spec.CredentialsSecretRef.Key], p.Spec.Region, awsclients.UseProviderSecret)
+	return &external{client: awsClient, kube: c.client}, errors.Wrap(err, errNewClient)
+}
+
+type external struct {
+	client elasticache.Client
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.CacheCluster)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotCacheCluster)
+	}
+
+	resp, err := e.client.DescribeCacheClustersRequest(elasticache.NewDescribeCacheClustersInput(meta.GetExternalName(cr))).Send(ctx)
+	if err != nil {
+		return managed.ExternalObservation{ResourceExists: false}, errors.Wrap(resource.Ignore(elasticache.IsClusterNotFound, err), errDescribeCacheCluster)
+	}
+
+	cluster := resp.CacheClusters[0]
+	current := cr.Spec.ForProvider.DeepCopy()
+	elasticache.LateInitializeCluster(&cr.Spec.ForProvider, cluster)
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errUpdateCacheClusterCR)
+		}
+	}
+
+	cr.Status.AtProvider = elasticache.GenerateClusterObservation(cluster)
+
+	switch cr.Status.AtProvider.CacheClusterStatus {
+	case v1alpha1.StatusAvailable:
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+	case v1alpha1.StatusCreating:
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	case v1alpha1.StatusDeleting:
+		cr.Status.SetConditions(runtimev1alpha1.Deleting())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Unavailable())
+	}
+
+	upToDate, err := elasticache.IsClusterUpToDate(meta.GetExternalName(cr), &cr.Spec.ForProvider, &cluster)
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: upToDate,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.CacheCluster)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotCacheCluster)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	_, err := e.client.CreateCacheClusterRequest(elasticache.GenerateCreateCacheClusterInput(cr.Spec.ForProvider, meta.GetExternalName(cr))).Send(ctx)
+
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreateCacheCluster)
+}
+
+func (e *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.CacheCluster)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotCacheCluster)
+	}
+
+	// AWS API rejects modification requests if the state is not `available`
+	if cr.Status.AtProvider.CacheClusterStatus != v1beta1.StatusAvailable {
```

Comment:
I noticed in the switch block in `Observe` that this is compared to `v1alpha1.StatusAvailable` but here is compared to `v1beta1.StatusAvailable`. I am guessing the former is what you want?